core/domain/stats_jobs_one_off_test.py

```diff
@@ -812,6 +896,52 @@ def test_recompute_for_version_with_no_events(self):
         model = stats_models.ExplorationStatsModel.get(model_id)
         self.assertEqual(model.num_actual_starts_v2, 3)
 
+    def test_job_with_no_exploration_version(self):
+        model1 = stats_models.ExplorationActualStartEventLogEntryModel.get(
+            'id1')
+        model2 = stats_models.ExplorationActualStartEventLogEntryModel.get(
+            'id2')
+        model3 = stats_models.ExplorationActualStartEventLogEntryModel.get(
+            'id3')
+        model4 = stats_models.ExplorationActualStartEventLogEntryModel.get(
+            'id4')
+        stats_models.ExplorationActualStartEventLogEntryModel.delete_multi(
+            [model1, model2, model3, model4])
+        stats_models.ExplorationActualStartEventLogEntryModel(
+            id='id',
+            exp_id=self.EXP_ID,
+            exp_version=None,
+            state_name=self.STATE_NAME,
+            session_id='session_id_4',
+            event_schema_version=2).put()
+
+        job_output = self.run_one_off_job()
+        self.assertEqual(
+            job_output, ['[u\'None version for EXP_ID actual_start state_1\']'])
+
+    def test_job_with_invalid_exploration_id(self):
+        stats_models.ExplorationActualStartEventLogEntryModel(
+            id='id',
+            exp_id='invalid_exp_id',
+            exp_version=self.EXP_VERSION,
+            state_name=self.STATE_NAME,
+            session_id='session_id_4',
+            event_schema_version=2).put()
+
+        job_output = self.run_one_off_job()
+        self.assertEqual(job_output, [])
```

Comment:
If it's invalid, why are no error messages returned?

Reply:
We are not returning any error messages due to invalid exp ids. There will be error messages only if there are missing stats models or if the exp version is None. In case of an invalid exp id, we are returning an empty list. See here:
https://github.com/oppia/oppia/blob/574b643cf8f1dea6e93d1471c57a019af32cbfcb/core/domain/stats_jobs_one_off.py#L454

Reply:
Done